#! /bin/sh /usr/share/dpatch/dpatch-run
## 14-xmalloc.dpatch by Sven Mueller <debian@incase.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix xmalloc usage

@DPATCH@
--- cyrus-imapd.orig/lib/map_stupidshared.c
+++ cyrus-imapd/lib/map_stupidshared.c
@@ -47,6 +47,7 @@
 #include <sys/stat.h>
 #include <syslog.h>
 
+#include "xmalloc.h"
 #include "map.h"
 #include "exitcodes.h"
 #include "xmalloc.h"
--- cyrus-imapd.orig/lib/xmalloc.h
+++ cyrus-imapd/lib/xmalloc.h
@@ -60,7 +60,7 @@ extern void *xmemdup (const void *ptr, u
 
 /* Functions using xmalloc.h must provide a function called fatal() conforming
    to the following: */
-extern void fatal(const char *fatal_message, int fatal_code);
-/*   __attribute__ ((noreturn));*/
+extern void fatal(const char *fatal_message, int fatal_code)
+   __attribute__ ((noreturn));
 
 #endif /* INCLUDED_XMALLOC_H */
